File: first_tries/python_2_7/calendarSVG.py
#calendarSVG.py

import datetime
import calendar


year = 2018
month = 4


viewBoxWidth = 2550
viewBoxHeight = 3300
radius = viewBoxWidth/16
day_xmargin = radius/10
day_ymargin = radius
centerX = radius
centerY = radius
outline_style_for_day = "fill:rgb(200,200,255);"
font_size_for_day = int(radius)
font_style_for_day = "font-family:\'Helvetica\';font-size:%spx;fill:rgb(102,102,102);" % font_size_for_day
font_size_for_month = int(2*radius)
font_style_for_month = "font-family:\'Helvetica\';font-size:%spx;fill:rgb(102,102,102);" % font_size_for_month

first_day = datetime.date(year, month, 1)
first_weekday = first_day.strftime("%w") #Sunday = 0
first_week_of_the_month = first_day.strftime("%U") #Sunday = 0
file_name = first_day.strftime("%Y%B")
month_name = first_day.strftime("%B, %Y")
day_range_of_the_month = calendar.monthrange(year,month)

f = open('%s.svg' % file_name, 'w')
#print month_name
f.write('<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n')
f.write('<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n')
f.write('<svg width="100%%" height="100%%" viewBox="0 0 %s %s" xmlns="http://www.w3.org/2000/svg">\n' % (viewBoxWidth, viewBoxHeight))

f.write('\t<g transform="translate(%s, %s)" style="%s">\n' % ((radius/2), (viewBoxHeight/4), outline_style_for_day))
for x in range(0, day_range_of_the_month[1]):
    my_date = datetime.date(year, month, x+1)
    my_weekday = int(my_date.strftime("%w")) #Sunday = 0
    my_week = int(int(my_date.strftime("%U")) - int(first_week_of_the_month))
    #print ("my day: %s, my week: %s, my day: %s") % (x+1, my_week, my_weekday)
    my_x = centerX + day_xmargin + (my_weekday * ((radius*2)+day_xmargin))
    my_date_x = my_x + day_xmargin/2
    my_y = centerY + (my_week * ((radius*2)+day_ymargin))
    my_date_y = my_y #- font_size_for_day
    f.write('\t<circle cx="%s" cy="%s" r="%s"/>\n' % (my_x, my_y, radius))
    f.write('\t<text x="%s" y="%s" style="%s" text-anchor="middle" dominant-baseline="central">%s</text>\n' % (my_date_x, my_date_y, font_style_for_day, my_date.strftime("%d")))
f.write('\t</g>\n')


f.write('\t<g transform="translate(%s, %s)" style="%s">\n' % ((radius/2), (viewBoxHeight/8), font_style_for_month))
f.write('\t\t<text>%s</text>\n' % month_name)
f.write('\t</g>\n')

f.write('</svg>')
f.close()

File: first_tries/javascript/calendar_svg/calendar_svg.js
const fs = require('fs')
let moment = require('moment')
//https://momentjs.com/docs/#/displaying/

let year = 2018
let month = 3

let viewBoxWidth = 2550
let viewBoxHeight = 3300
let radius = viewBoxWidth/16
let day_xmargin = radius/10
let day_ymargin = radius
let centerX = radius
let centerY = radius
let outline_style_for_day = "fill:rgb(200,200,255);"
let font_size_for_day = radius
let font_style_for_day = `font-family:\'Helvetica\';font-size:${font_size_for_day}px;fill:rgb(102,102,102);`
let font_size_for_month = 2*radius
let font_style_for_month = `font-family:\'Helvetica\';font-size:${font_size_for_month}px;fill:rgb(102,102,102);`


let day = new Date(year, month, 1);
let first_day = moment(day);
first_weekday = first_day.format('d')
first_week_of_the_month = first_day.format('w')
file_name = first_day.format('YYYYMMM')
month_name = first_day.format("MMMM, YYYY")
last_day_of_the_month = first_day.daysInMonth()
console.log(`START !!! my date: ${day} my week: ${first_week_of_the_month}, my day: ${first_weekday}"`)


let svg = ""
svg += '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n'
svg += '<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n'

svg += `<svg width="100%" height="100%" viewBox="0 0 ${viewBoxWidth} ${viewBoxHeight}" xmlns="http://www.w3.org/2000/svg">\n`

svg += `\t<g transform="translate(${(radius/2)}, ${(viewBoxHeight/4)})" style="${outline_style_for_day}">\n`

for (let x = 0; x < last_day_of_the_month; x++) {
    let my_date = new Date(year, month, x+1)
    let my_moment = moment(my_date)
    let my_weekday = my_moment.format('d')
    let my_week = my_moment.format('w') - first_week_of_the_month
    //console.log(`my date: ${my_date} my week: ${my_week}, my day: ${my_weekday}"`)
    let my_x = centerX + day_xmargin + (my_weekday * ((radius*2)+day_xmargin))
    let my_date_x = my_x + day_xmargin/2
    let my_y = centerY + ((my_week) * ((radius*2)+day_ymargin))
    let my_date_y = my_y //- font_size_for_day
    let my_date_text = my_moment.format('DD')

    svg += `\t<circle cx="${my_x}" cy="${my_y}" r="${radius}"/>\n`
    svg += `\t<text x="${my_date_x}" y="${my_date_y}" style="${font_style_for_day}" text-anchor="middle" dominant-baseline="central">${my_date_text}</text>\n`


    //svg += `\t<circle cx="${centerX+(x*((radius*2)+5))}" cy="${centerY}" r="${radius}" style="${outline_style_for_day}"/>\n`
}
svg += '\t</g>\n'

svg += `\t<g transform="translate(${(radius/2)}, ${(viewBoxHeight/8)})" style="${font_style_for_month}">\n`
svg += `\t\t<text>${month_name}</text>\n`
svg += '\t</g>\n'

svg += '</svg>'

fs.writeFile(`${file_name}.svg`, svg, (err) => {  
    // throws an error, you could also catch it here
    if (err) throw err;

    // success case, the file was saved
    console.log('SVG !');
});
